src/xcode/ENA/ENA/Source/AppDelegate & Globals/ENATaskExecutionDelegate.swift

```diff
@@ -224,4 +237,9 @@ class TaskExecutionHandler: ENATaskExecutionDelegate {
 
 		riskProvider.requestRisk(userInitiated: false)
 	}
+
+	private func executeAnalyticsSubmission(completion: @escaping () -> Void) {
+		self.analyticsSubmitter.triggerSubmitData()
+		completion()
```

Comment:
Since triggerSubmitData is an async process, I'm concerned that it will be interrupted if the completion is called directly and it leads to completion of the background process.

Reply:
yeah, I will add the completion handler here